src/core_modules/capture-core/components/DataEntries/SingleEventRegistrationEntry/DataEntryWrapper/DataEntry/DataEntry.component.js

```diff
@@ -567,8 +572,9 @@ class NewEventDataEntry extends Component<Props> {
                     {/* $FlowFixMe[cannot-spread-inexact] automated comment */}
                     <WrappedDataEntry
                         id={'singleEvent'}
-                        onUpdateFormField={onUpdateField}
-                        onUpdateFormFieldAsync={onStartAsyncUpdateField}
+                        onUpdateDataEntryField={onUpdateDataEntryField(orgUnit)}
+                        onUpdateFormField={onUpdateField(orgUnit)}
+                        onUpdateFormFieldAsync={onStartAsyncUpdateField(orgUnit)}
```

Comment:
I did not consider performance when I wrote this. These props are passed at least a few levels I think, so it would quite clearly improve the efficiency to have these functions wrapped inside useMemos. We're dealing with a class component here however, using `this` and a couple of lifecycle methods. Not feeling too confident about getting the rewrite into a function component right.